File: MCD_Python/mtg_card_detector/models/card.py
"""
Classes for representing card candidates and recognized cards.
"""

import numpy as np
from dataclasses import dataclass
from shapely.geometry.polygon import Polygon


@dataclass
class CardCandidate:
    """
    Class representing a segment of the image that may be a recognizable card.
    """
    image: np.ndarray
    bounding_quad: Polygon
    image_area_fraction: float
    is_recognized: bool = False
    recognition_score: float = 0.
    is_fragment: bool = False
    name: str = 'unknown'

    def contains(self, other):
        """
        Returns whether the bounding polygon of the card candidate
        contains the bounding polygon of the other candidate.
        """
        return bool(other.bounding_quad.within(self.bounding_quad) and
                    other.name == self.name)

File: mcd_dart/lib/models/card.dart
// Import when needed
// import 'dart:typed_data';
import 'package:image/image.dart';
import '../geometry/polygons.dart';

class CardCandidate {
  final Image image;
  final Polygon boundingQuad;
  final double imageAreaFraction;
  bool isRecognized;
  double recognitionScore;
  bool isFragment;
  String name;

  CardCandidate(
    this.image,
    this.boundingQuad,
    this.imageAreaFraction, {
    this.isRecognized = false,
    this.recognitionScore = 0.0,
    this.isFragment = false,
    this.name = 'unknown',
  });

  bool contains(CardCandidate other) {
    return boundingQuad.contains(other.boundingQuad) && name == other.name;
  }
}